File: ios/ShortDrama/Sources/Domain/UseCases/CreateCommentUseCase.swift
import Foundation

struct CreateCommentUseCase: Sendable {
    private let repository: CommentRepositoryProtocol

    init(repository: CommentRepositoryProtocol) {
        self.repository = repository
    }

    func execute(dramaId: String, content: String) async throws -> Comment {
        try await repository.createComment(dramaId: dramaId, content: content)
    }
}

File: android/app/src/main/java/com/djs66256/short_drama/domain/usecase/CreateCommentUseCase.kt
package com.djs66256.short_drama.domain.usecase

import com.djs66256.short_drama.core.network.ApiResult
import com.djs66256.short_drama.domain.model.Comment
import com.djs66256.short_drama.domain.repository.CommentRepository
import javax.inject.Inject

class CreateCommentUseCase @Inject constructor(
    private val commentRepository: CommentRepository,
) {
    suspend operator fun invoke(
        dramaId: String,
        content: String,
    ): ApiResult<Comment> {
        return commentRepository.createComment(dramaId, content)
    }
}
